### ISSUE TITLE
Polish Visualization test suite

### ISSUE THREAD

**Issue by aherzberg** [department-of-veterans-affairs/va-mobile-app#6796]

**Comment by timwright12** [comment]
@aherzberg @StacyB2023 is this still something we're working on?

**Comment by StacyB2023** [comment]
@timwright12 Based on Stand Ups, Im going to say No. I've not seen this. @aherzberg Thoughts on this one?, I may need to get with you to understand what this effort was about. Pending Handoff tag looks to be 10 months ago.

**Comment by timwright12** [comment]
cool, if this is a dead project lets wind it down, close out tickets, and evaluate if the code should be deleted/archived

**Comment by aherzberg** [comment]
I think I made this as more of a placeholder since the the FE was still pretty barebones and could use some work to make it look more professional. In terms of next steps for this project, we'd need this to get this hosted on a central server. Right now it's running periodically on each of our computers. We'd also need to add a slack integration or email client to notify us when something goes down. I think that could still have some value.

**Comment by timwright12** [comment]
Can we bring @donmccaughey into this to see if we should move forward?

**Comment by aherzberg** [comment]
Yeah! lets do it. I can make a quick video of the site

**Comment by timwright12** [comment]
wrong account ping, sorry @DonMcCaugheyUSDS

**Comment by aherzberg** [comment]
[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeU1JQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bcf3f69764b14eea47e8cd4e2f6bc596a275f3c9/Screen%20Recording%202024-07-24%20at%2011.57.35%E2%80%AFAM.mov)

Here's a quick screenshare of the state of the test suite. Note that the development environment has changes as well as some of the staging users data and/or passwords since we last worked on this so it looks like a good number of tests are failing due to this. (the endpoints in real life though are in good health). If we decide to continue work on this project then I'll make a ticket to look into this issues but didn't want to take the time to fix it if we wanted to discontinue it. Right now isn't bringing any value to us but I think that could change if we put it on a central server and create a slack integration. Could also use a cosmetic facelift.

**Comment by jperk51** [comment]
@timwright12 @DonMcCaugheyUSDS Let us know if there's interest in continuing to work on this

**Comment by DonMcCaugheyUSDS** [comment]
I can't see the video so working from a limited understanding of what this is.

My high-level questions:
- Is this catching any errors or bugs for us?
- If so, are those bugs severe (i.e. preventing veterans from completing tasks)?
- Do we have an estimate for the amount of work to get all the tests passing and keep them there?

Basically, in your respective judgements, do we get benefits out of this test suite that justifies putting time and effort into it?

**Comment by jperk51** [comment]
The main purpose of this was to have something frequently hitting staging so if a service was down we could maybe get team responsible to fix it before QA starting testing something related to it. That being said I haven't heard anything from QA about running into this issue anymore so it's very possible that staging is just better now and/or we have just become better at knowing where to look for issues/planned downtimes since we put this into place. The UI was just meant to have a nice place for non-technical people to see a service map that indicated current health. I don't think _we_ get a ton of use out of it because we have datadog and stuff that answers the same questions for us, but I could see some potential interest from other groups although idk of any interest currently. 

@TKDickson Have you or anyone else in QA run into staging being busted unexpectedly like we used to frequently in the past?

@aherzberg How much work do you think it would take to update the tests/UI to better match VA styles? Also do you think it's possible to achieve something similar using datadog/actioner/slack by leaving this as just something that automatically makes the calls but doesn't worry about handling the testing?

**Comment by TKDickson** [comment]
@jperk51 the QA experience of staging is much more stable -- a lot of that can be attributed to the improvements we made on the mobile team, when changing over to the v2/user call, where we did a much better job of isolating API issues to only impact the pages where we needed that info. A resource for QA that would help answer the question "is this endpoint universally down, or is it just a specific user/data issue that I'm running into" would still be helpful, but 1) I think the visualizer is much larger than that need currently, and 2) I'd downgrade whatever we make for that need to a 'nice to have' given the improvements we've made.

lmk if you've got other questions.
